Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -779,13 +779,39 @@ public struct RenderNodeTranslator: SemanticVisitor {
             
             return seeAlsoSections
         } ?? .init(defaultValue: [])
+
+        if let callToAction = article.metadata?.callToAction {
+            if let url = callToAction.url {
+                let downloadIdentifier = RenderReferenceIdentifier(url.description)
+                node.sampleDownload = .init(
+                    action: .reference(
+                        identifier: downloadIdentifier,
+                        isActive: true,
+                        overridingTitle: callToAction.buttonLabel(),
```

Comment:
What happens if we don't provide an overriding title? Should we be allowing the renderer to pick the title in some cases?

Reply:
The `buttonLabel` method (now property) checks the `label` parameter as well as the `purpose` parameter, so the "default" label is determined by what `purpose` an author writes. We could make these parameters totally optional and default all the way down to "Download", but i think forcing the `purpose` parameter is good enough.